src/mudlet-lua/lua/GUIUtils.lua

```diff
@@ -2024,6 +2024,7 @@ end
 function ansi2decho(text, ansi_default_color)
   assert(type(text) == 'string', 'ansi2decho: bad argument #1 type (expected string, got '..type(text)..'!)')
   local lastColour = ansi_default_color
+  local coloursToUse = nil
```

Comment:
This wasn't working because `coloursToUse` was being initialized inside the loop and thus its value was being reset on each iteration.